## 1. Introduzca en R la siguiente matriz $A$ 4×3 usando:

In [1]:
A = matrix(c(1,2,3,4,5,6,7,8,9,10,11,12), nrow=4,"byrow"="true")
A

[,1] [,2] [,3]
[1,]  1    2    3  
[2,]  4    5    6  
[3,]  7    8    9  
[4,] 10   11   12

Luego, acceda a los algunos elementos de la matriz de la siguiente manera: A[1,1:3], A[1:4,2],
A[3,3], A[11], A[20], A[5,4], A[1,1,1] y explique qué pasa en cada caso.

1. En este caso obtenemos el renglón 1, y las columnas de la uno a la 3

In [2]:
A[1,1:3]

[1] 1 2 3

2. Aquí tenemos del renglón 1 al 4, de la columna 2 

In [3]:
A[1:4,2]

[1]  2  5  8 11

3. Aquí el elemento del renglón 3, columna 3

In [4]:
A[3,3]

[1] 9

3. Al utilizar solamente un índice, obtenemos el onceavo elemento leyendo por columnas tal somo se muestra a continuación

$$\begin{bmatrix}
a_1 & a_5 & a_9\\
a_2 & a_6 & a_{10}\\ 
a_3 & a_7 & a_{11}\\
a_4 & a_8 & a_{12}
 \end{bmatrix} = \begin{bmatrix}
1 & 2 & 3\\
4 & 5 & 6\\ 
7 & 8 & {\boldsymbol 9}\\
10 & 11 & 12
 \end{bmatrix}$$

In [5]:
A[11]

[1] 9

4. En este caso, no existe un elemento correspondiente al 20vo elemento, por lo que regresa <NA>

In [6]:
A[20]

[1] NA

5. Recordando que nuetra matriz es de la forma, no existe fila 5 columna 4, por lo que obtenemos un error de "out of bounds"

$$\begin{bmatrix}
1 & 2 & 3\\
4 & 5 & 6\\ 
7 & 8 & {\boldsymbol 9}\\
10 & 11 & 12
 \end{bmatrix}$$

In [7]:
A[5,4]

ERROR: Error in A[5, 4]: subscript out of bounds


En este caso nuestra matriz es de $4 \times 3$. Necesitariamos una matrix de 3 dimensiones $n \times m \times p$ con ${n,m,p} \in \mathbb{Z}{+}$

In [8]:
A[1,1,1] 

ERROR: Error in A[1, 1, 1]: incorrect number of dimensions


## 2. Investigue para qué sirven los comandos de R, as.matrix(...) y as.data.frame(...), explique y dé un ejemplo de cada uno.

Primero vermos as.data.frame

In [9]:
m1<-matrix(c(1,2,3,4,5,6,7,8,9),nrow=3, ncol=3, byrow = T)
m1


[,1] [,2] [,3]
[1,] 1    2    3   
[2,] 4    5    6   
[3,] 7    8    9

In [10]:
dataframe<-as.data.frame(m1)
dataframe

V1 V2 V3
1 1  2  3 
2 4  5  6 
3 7  8  9

In [11]:
class(dataframe)

[1] "data.frame"

### Vemos que as.data.frame(m1) convierte la matriz m1 en un dataframe

Ahora veremos que hace as.matrix

In [12]:
a_matrix <- as.matrix(dataframe)
a_matrix

V1 V2 V3
[1,] 1  2  3 
[2,] 4  5  6 
[3,] 7  8  9

In [13]:
class(a_matrix)

[1] "matrix"

### Con esto podemos ver que as.matrix convierte un dataframe en una matrix

## 3. Introduzca usando código R (no archivos) en un DataFrame la siguiente tabla de datos:

In [14]:
# Vamos a definir la variable categórica nivel_educativo
nivel_educativo<-factor(c("Lic","Bach","Bach","Bach","Dr","MSc"), levels=c("Lic","Bach","Dr","MSc"))
nivel_educativo 

[1] Lic  Bach Bach Bach Dr   MSc 
Levels: Lic Bach Dr MSc

In [15]:
## Se define la hoja de datos mediante:

datos<-data.frame(Peso=c(76,67,55,57,87,48),Edad=c(25,23,19,18,57,13),"Nivel Educativo"=nivel_educativo)
datos

Peso Edad Nivel.Educativo
1 76   25   Lic            
2 67   23   Bach           
3 55   19   Bach           
4 57   18   Bach           
5 87   57   Dr             
6 48   13   MSc

## 4. Cargue en un DataFrame el archivo “Ventas.csv” y haga lo siguiente:

In [16]:
## Vamos a establecer el directorio
setwd("/home/jovyan/work/tratamiento")

## Luego para poner la Tabla de datos Ventas.csv en un "DataFrame" se usa función read.table
Datos<-read.table("Ventas.csv",header=TRUE, sep=",",dec=".",colClasses = c("NumeroTienda"="factor"))
head(Datos)

Mes       NumeroTienda Zona   ImporteVenta Pago     Edad
1 AGOSTO    2            Centro 515.75       Aplazado 57  
2 ENERO     5            Sur    549.47       Contado  29  
3 AGOSTO    4            Centro 544.78       Contado  41  
4 MARZO     6            Sur    522.36       Aplazado 63  
5 DICIEMBRE 2            Centro 532.33       Contado  44  
6 FEBRERO   2            Centro 524.87       Contado  34

o Calcule la dimensión de la tabla de datos ( R= $102 \times 6$ )

In [17]:
dim(Datos)

[1] 102   6

o Despliegue las primeras 2 columnas de la tabla de datos. Usaremos head() para solo desplegar las primeras 6 filas

In [18]:
head(Datos[,1:2])

Mes       NumeroTienda
1 AGOSTO    2           
2 ENERO     5           
3 AGOSTO    4           
4 MARZO     6           
5 DICIEMBRE 2           
6 FEBRERO   2

o Ejecute un “summary” y un “str” de los datos.

In [19]:
str(Datos)

'data.frame':	102 obs. of  6 variables:
 $ Mes         : Factor w/ 12 levels "ABRIL","AGOSTO",..: 2 4 2 8 3 5 1 8 4 12 ...
 $ NumeroTienda: Factor w/ 9 levels "1","2","3","4",..: 2 5 4 6 2 2 7 5 3 7 ...
 $ Zona        : Factor w/ 3 levels "Centro","Norte",..: 1 3 1 3 1 1 2 3 1 2 ...
 $ ImporteVenta: num  516 549 545 522 532 ...
 $ Pago        : Factor w/ 2 levels "Aplazado","Contado": 1 2 2 1 2 2 2 2 2 1 ...
 $ Edad        : int  57 29 41 63 44 34 50 45 21 43 ...


In [20]:
summary<-summary(Datos)
summary

         Mes      NumeroTienda     Zona     ImporteVenta         Pago   
 DICIEMBRE :14   5      :16    Centro:38   Min.   :500.6   Aplazado:32  
 JUNIO     :10   6      :15    Norte :33   1st Qu.:513.8   Contado :70  
 MARZO     :10   3      :14    Sur   :31   Median :523.1                
 SEPTIEMBRE:10   7      :13                Mean   :525.5                
 MAYO      : 9   9      :11                3rd Qu.:541.1                
 AGOSTO    : 8   2      : 9                Max.   :555.9                
 (Other)   :41   (Other):24                                             
      Edad      
 Min.   :21.00  
 1st Qu.:30.00  
 Median :39.00  
 Mean   :38.74  
 3rd Qu.:45.00  
 Max.   :65.00  
                

### Las únicas columnas de las que tiene sentido obtener Media y Desviación Estandar (no categóricas), son ImporteVenta y Edad

La media ya la tenemos gracias a summary

- ImporteVenta = 525.5
- Edad = 38.74

La desviación estándard la calcularemos haciendo uso de la función $\texttt{sd()}$, obteniendo:

- ImporteVenta = 15.75
- Edad = 10.75

In [21]:
sd(Datos$ImporteVenta)

[1] 15.75831

In [22]:
sd(Datos$Edad)

[1] 10.75738

### Calculando " a mano " la media

In [23]:
media_importe_venta<-sum(Datos$ImporteVenta)/length(Datos$ImporteVenta)
media_importe_venta

[1] 525.4526

In [24]:
media_edad<-sum(Datos$Edad)/length(Datos$Edad)
media_edad

[1] 38.73529

### Calculando " a mano " la desviación estandard

$$\sigma = \sqrt{\frac{\sum(x_i-\mu)^2}{N-1}}$$

In [25]:
sqrt(sum((Datos$ImporteVenta - media_importe_venta)^2)/(length(Datos$ImporteVenta)-1))

[1] 15.75831

In [26]:
sqrt(sum((Datos$Edad - media_edad)^2)/(length(Datos$Edad)-1))

[1] 10.75738

## Calcule la suma de la columna ImporteVenta.

In [27]:
sum(Datos$ImporteVenta)

[1] 53596.17

## Calcule en R para todas las columnas de tipo cuantitativo en el archivo: El mínimo, el máximo, la media, la mediana.

### Haciendo uso de las funciones integradas $\texttt{min()}$,  $\texttt{max()}$ y $\texttt{median()}$

In [28]:
min(Datos$ImporteVenta)

[1] 500.59

In [29]:
 max(Datos$ImporteVenta)

[1] 555.85

In [30]:
media_importe_venta

[1] 525.4526

In [32]:
median(Datos$ImporteVenta)

[1] 523.14

### Ahora para Edad

In [34]:
min(Datos$Edad)

[1] 21

In [35]:
max(Datos$Edad)

[1] 65

In [37]:
media_edad

[1] 38.73529

In [38]:
median(Datos$Edad)

[1] 39

### Para la variable pago determine las cuantos Aplazados y cuantos de Contado, hay en la tabla.

In [40]:
table(Datos$Pago)


Aplazado  Contado 
      32       70 